presto-main/src/main/java/com/facebook/presto/sql/planner/RelationPlanner.java

```diff
@@ -771,7 +771,7 @@ protected RelationPlan visitValues(Values node, SqlPlannerContext context)
         ImmutableList.Builder<List<RowExpression>> rowsBuilder = ImmutableList.builder();
         for (Expression row : node.getRows()) {
             ImmutableList.Builder<RowExpression> values = ImmutableList.builder();
-            if (row instanceof Row) {
+            if (row instanceof Row && !scope.requireSingleRowValues()) {
```

Comment:
Why is single row a special case?

Reply:
If there is only one target column, statement `insert into <table> values (a), (b)` and `insert into <table> values a, b` would do the same thing. And they will be parsed to the same `Statement` in which the values are not wrapped in a `Row`. That means, if the unary column's type is `row(a int, b varchar)`, it will be confused with two columns of `int` and `varchar` without knowing the column type pre-read from table metadata.

So, we should decide whether to unfold a `Row` base on value `scope.requireSingleRowValues()`.